tests/ilios_test.php

```diff
@@ -61,19 +62,21 @@ public function test_get_schools(): void {
         set_config('host_url', 'http://ilios.demo', 'tool_ilioscategoryassignment');
 
         $handlerstack = HandlerStack::create(new MockHandler([
-            function(RequestInterface $request) {
-                $this->assertEquals('/api/v3/schools', $request->getUri()->getPath());
-                return new Response(200, [], json_encode([
+                new Response(200, [], json_encode([
                     'schools' => [
                         ['id' => 1, 'title' => 'Medicine'],
                         ['id' => 2, 'title' => 'Pharmacy'],
                     ],
-                ]));
-            },
+                ])),
         ]));
+        $container = [];
+        $history = Middleware::history($container);
+        $handlerstack->push($history);
         di::set(http_client::class, new http_client(['handler' => $handlerstack]));
         $ilios = di::get(ilios::class);
         $schools = $ilios->get_schools();
+
+        $this->assertEquals('/api/v3/schools', $container[0]['request']->getUri()->getPath());
```

Comment:
I do like this better than having it in the callback above where the transaction happens, that is a nice improvement.